### PR TITLE
Changing classic hymn titles to include first line of verse

### PR DESCRIPTION
**Pull request by rachely3n**
Addressing #186. I will squash my commits before merging.

Note that the title ends in punctuation since a lot of the lyric lines end in some sort of punctuation. Is that alright or should we strip it?
Screen Shot 2020-05-30 at 6 23 34 PM

### CODE REVIEW

**Comment by rachely3n** [comment]
Don't understand why test passes for me locally but fails here...

**Comment by skywalkerdude** [comment]
they're flaky :(

File: Hymns/Display/DisplayHymnViewModel.swift [hunk 1]
```diff
@@ -66,7 +66,13 @@ class DisplayHymnViewModel: ObservableObject {
 
                     let title: Title
                     if self.identifier.hymnType == .classic {
-                        title = "Hymn \(self.identifier.hymnNumber)"
+                        var titleLyric = ""
+                        if let lyrics = hymn.lyrics.first {
+                            if let firstLine = lyrics.verseContent.first {
+                                titleLyric = ": \(firstLine)"
+                            }
+                        }
```
Comment (skywalkerdude) [inline Hymns/Display/DisplayHymnViewModel.swift:74]: I don't think you need to do this. you have access to hymn.title, and you can just use that.

So it'd be like
```
let title = ""
if hymnType == classic {
  title = "Hymn \(hymnumber): "
}
title.append(hymn.title.replacingOccurrences(of: "Hymn: ", with: ""))
```

File: HymnsTests/Display/DisplayHymnViewModelSpec.swift [hunk 2]
```diff
@@ -54,7 +54,8 @@ class DisplayHymnViewModelSpec: QuickSpec {
                         beforeEach {
                             target = DisplayHymnViewModel(backgroundQueue: testQueue, favoritesStore: favoritesStore, hymnToDisplay: classic1151, hymnsRepository: hymnsRepository, historyStore: historyStore,
                                                           mainQueue: testQueue, pdfPreloader: pdfLoader, storeInHistoryStore: true)
-                            let hymn = UiHymn(hymnIdentifier: classic1151, title: "title", lyrics: [Verse](), pdfSheet: Hymns.MetaDatum(name: "Lead Sheet", data: [Hymns.Datum(value: "Piano", path: "/en/hymn/c/1151/f=ppdf"), Hymns.Datum(value: "Guitar", path: "/en/hymn/c/1151/f=pdf"), Hymns.Datum(value: "Text", path: "/en/hymn/c/1151/f=gtpdf")]))
+                            let hymn = UiHymn(hymnIdentifier: classic1151, title: "title", lyrics: [Verse(verseType: VerseType.verse, verseContent: ["Drink! A river pure and clear that’s flowing from the throne;"])], pdfSheet: Hymns.MetaDatum(name: "Lead Sheet", data: [Hymns.Datum(value: "Piano", path: "/en/hymn/c/1151/f=ppdf"), Hymns.Datum(value: "Guitar", path: "/en/hymn/c/1151/f=pdf"), Hymns.Datum(value: "Text", path: "/en/hymn/c/1151/f=gtpdf")]))
```
Comment (skywalkerdude) [inline HymnsTests/Display/DisplayHymnViewModelSpec.swift:57]: Then you can revert this and just use the fake `title` instead

File: Hymns/Display/DisplayHymnViewModel.swift (hunk 1 above)
Comment (skywalkerdude) [reply Hymns/Display/DisplayHymnViewModel.swift:74]: I think that also takes care of your punctuation problem!

File: Hymns/Display/DisplayHymnViewModel.swift [hunk 3]
```diff
@@ -64,12 +64,13 @@ class DisplayHymnViewModel: ObservableObject {
                     guard let self = self else { return }
                     guard let hymn = hymn else { return }
 
-                    let title: Title
+                    var title: Title
```
Comment (skywalkerdude) [inline Hymns/Display/DisplayHymnViewModel.swift:67]: I think this can still be a let

Comment (rachely3n) [reply Hymns/Display/DisplayHymnViewModel.swift:67]: It complained later when I did the append since let made it immutable.

**Pull request merged**